src/commands/runDraftTool/runDraftDockerfile.ts

```diff
@@ -0,0 +1,330 @@
+   sourceCodeFolder: vscode.Uri;
+   language: DraftLanguage;
+   version: string;
+   port: string;
+}
+type WizardContext = IActionContext & Partial<PromptContext>;
+type IPromptStep = AzureWizardPromptStep<WizardContext>;
+type IExecuteStep = AzureWizardExecuteStep<WizardContext>;
+
+export async function runDraftDockerfile(
+   {actionContext, extensionContext}: Context,
+   sourceCodeFolder: vscode.Uri | undefined
+) {
+   const state: StateApi = State.construct(extensionContext);
+
+   // Ensure Draft Binary
+   const downloadResult = await longRunning(`Downloading Draft.`, () =>
+      downloadDraftBinary()
+   );
+   if (!downloadResult) {
+      vscode.window.showErrorMessage('Failed to download Draft');
+      return undefined;
+   }
+
+   const workspaceFolders = vscode.workspace.workspaceFolders;
+   if (!sourceCodeFolder && workspaceFolders && workspaceFolders.length !== 0) {
+      sourceCodeFolder = workspaceFolders[0].uri;
+   }
+   const wizardContext: WizardContext = {
+      ...actionContext,
+      sourceCodeFolder: sourceCodeFolder
+   };
+   const promptSteps: IPromptStep[] = [
+      new PromptSourceCodeFolder(),
+      new PromptDockerfileOverride(),
+      new PromptLanguage(),
+      new PromptVersion(),
+      new PromptPort()
+   ];
+   const executeSteps: IExecuteStep[] = [
+      new ExecuteDraft(),
+      new ExecuteOpenDockerfiles(),
+      new ExecuteSaveState(state),
+      new ExecutePromptAcr()
+   ];
+   const wizard = new AzureWizard(wizardContext, {
+      title,
+      promptSteps,
+      executeSteps
+   });
+   await wizard.prompt();
+   await wizard.execute();
+}
+
+class PromptSourceCodeFolder extends AzureWizardPromptStep<WizardContext> {
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      const sourceCodeFolder = (
+         await wizardContext.ui.showOpenDialog({
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+            stepName: 'Source Code Folder',
+            openLabel: 'Choose Source Code Folder',
+            title: 'Choose Source Code Folder',
+            defaultUri: wizardContext.sourceCodeFolder
+         })
+      )[0];
+
+      if (!vscode.workspace.getWorkspaceFolder(sourceCodeFolder)) {
+         throw Error(
+            'Chosen Source Code Folder is not in current workspace. Please choose a folder in the workspace'
+         );
+      }
+
+      wizardContext.sourceCodeFolder = sourceCodeFolder;
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class PromptLanguage extends AzureWizardPromptStep<WizardContext> {
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      const languageToItem = (lang: DraftLanguage) => ({label: lang.name});
+      const languageOptions: vscode.QuickPickItem[] =
+         draftLanguages.map(languageToItem);
+      const languagePick = await wizardContext.ui.showQuickPick(
```

Comment:
again more of a question than feedback - is there a reason why we're not using detect language?

Reply:
We want to let Draft autodetect it. To do that we need the dry-run draft command (it's not implemented yet). Dry run will also spit out paths for where files will be created so it will knock out a birds with one stone.

Reply:
gotcha, makes sense!